### PR TITLE
Add HW4

### PR DESCRIPTION
**Pull request by DariaBoris**

### CODE REVIEW

**Comment by DariaBoris** [comment]
Приветики)
Посмотрела лекцию по строкам, мне показалось, что не оч сложно.
Сделала домашку.

**Comment by HeorhSemianenia** [comment]
Please call MR's appropriately, here it will be nice to see HW4 somewhere in the title

**Comment by DariaBoris** [comment]
I did it by accident. Then tried to change MR name but couldn't find out how to change it. Will try one more time.

**Comment by DariaBoris** [comment]
Found how to edit it.

File: hw4/Task_1.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+//Task 1
+// Пользователь передает через переменную свою фамилию и имя.
+// Выведите приветсвие `Привет <Имя> <Фамилия>` 5 раз используя для каждого варианта разный регистр
+
+const myNameSurnameUpper: string = 'Hey Daria Boris! '.toUpperCase();
```
Comment (HeorhSemianenia) [inline hw4/Task_1.ts]: "через переменную"

File: hw4/Task_1.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+const myNameSurnameLower: string = 'Hey Daria Boris! '.toLowerCase();
+
+console.log(myNameSurnameUpper.repeat(5),);
+console.log(myNameSurnameLower.repeat(5),);
```
Comment (HeorhSemianenia) [inline hw4/Task_1.ts]: you have 10 times

File: hw4/Task_10.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+//Task 10
+// Дано произвольное число с плавающей точкой.
+// Реализовать методы округления данного числа до ближайшего меньшего, большего аргумента и ближайшего целого числа.
+
+let myNumber2: number = 100.9290857209857;
```
Comment (HeorhSemianenia) [inline hw4/Task_10.ts]: could be const

File: hw4/Task_11.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+//Task 11
+// Дано произвольное число.
+// Реализовать метод возводящий число в степень. Степень возведения указывается как параметр к методу.
```
Comment (HeorhSemianenia) [inline hw4/Task_11.ts:3]: You should create separate custom method here, not use default one

File: hw4/Task_12.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+//Task 12
+// Реализовать метод получения рандомного целочисленного числа в заданном диапазоне.
+// Диапазон задается аргументами фунции
```
Comment (HeorhSemianenia) [inline hw4/Task_12.ts:3]: Also, create method/function here

File: hw4/Task_13.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+//Task 13
+// Реализовать метод получения рандомного числа с плавающей точкой в заданном диапазоне.
```
Comment (HeorhSemianenia) [inline hw4/Task_13.ts:2]: same

File: hw4/Task_14.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+//Task 14
+// Выведите дату сегоднешнего дня в формате: "текущая дата: месяц/год/день.
+// Текущее время часы:минуты:секунды.".
+// Использовать только внутренние методы Date.
+
+const date: Date = new Date();
+console.log('текущая дата: ' + date.toLocaleDateString()); // как вывести месяц/год/день?
```
Comment (HeorhSemianenia) [inline hw4/Task_14.ts]: https://stackoverflow.com/questions/27939773/tolocaledatestring-short-format

File: hw4/Task_3.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+//Task 3
+// Дана произвольная строка. Реализуйте метод, который принимает позицию символа в строке.
+// Если переданное число больше длины строки, то должно отобразиться сообщение:
```
Comment (HeorhSemianenia) [inline hw4/Task_3.ts:3]: method/function

File: hw4/Task_4.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+//Task 4
+// Дана произвольная строка. Реализуйте метод, который проверяет заканчивается ли предложение точкой или нет.
```
Comment (HeorhSemianenia) [inline hw4/Task_4.ts:2]: method/function

File: hw4/Task_5.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+//Task 5
+// Дана строка 'я учу typescript!'.
+// Вырежите из нее слово 'учу' и слово 'typescript' разными способами (через substring, slice).
+
+const myString1: string = 'Я учу TypeScript!';
+
+console.log(myString1)
+
+console.log(myString1.slice(2));
+console.log(myString1.substring(2))
```
Comment (HeorhSemianenia) [inline hw4/Task_5.ts]: image
Expected:
"Я TypeScript!"
"Я учу"

File: hw4/Task_9.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
+//Task 9
+// Дано произвольное число.
+// Реализовать метод проверяющий является ли это число целочисленным или нет.
```
Comment (HeorhSemianenia) [inline hw4/Task_9.ts:3]: method

File: hw4/Task_1.ts (hunk 1 above)
Comment (DariaBoris) [reply hw4/Task_1.ts]: Const myNameSurname created.

File: hw4/Task_1.ts (hunk 2 above)
Comment (DariaBoris) [reply hw4/Task_1.ts]: I repeated it 5 times for upper case and 5 for lower case.
Did I get task description wrong?

File: hw4/Task_10.ts (hunk 3 above)
Comment (DariaBoris) [reply hw4/Task_10.ts]: Ok, fixed.

File: hw4/Task_11.ts (hunk 4 above)
Comment (DariaBoris) [reply hw4/Task_11.ts:3]: I created a function.

File: hw4/Task_12.ts (hunk 5 above)
Comment (DariaBoris) [reply hw4/Task_12.ts:3]: I created a function.

File: hw4/Task_14.ts (hunk 7 above)
Comment (DariaBoris) [reply hw4/Task_14.ts]: Article helped, thanks. Fixed.

File: hw4/Task_3.ts (hunk 8 above)
Comment (DariaBoris) [reply hw4/Task_3.ts:3]: I created a function.

File: hw4/Task_4.ts (hunk 9 above)
Comment (DariaBoris) [reply hw4/Task_4.ts:2]: I created a function.

File: hw4/Task_5.ts (hunk 10 above)
Comment (DariaBoris) [reply hw4/Task_5.ts]: 1. I managed to get 'Я учу'.
2. I failed to get 'Я TypeScript!'. I cannot slice 'учу' while having 'Я TypeScript!' in my terminal. Need some help.

File: hw4/Task_9.ts (hunk 11 above)
Comment (DariaBoris) [reply hw4/Task_9.ts:3]: I created a function.

File: hw4/Task_11.ts (hunk 4 above)
Comment (HeorhSemianenia) [reply hw4/Task_11.ts:3]: LTGM

File: hw4/Task_3.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+// `Вы вышли за границу строки`
+
+function myFunction3(cat: string): string {
```
Comment (HeorhSemianenia) [inline hw4/Task_3.ts]: Please pay attention to naming of the functions and variables. They have to be more explicit and sensible.
For instance "myFunction3" -> "checkStingLength", "cat" -> "sentence".

File: hw4/Task_9.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+// Если является - вывести его, если нет - округлить до целой части.
+
+function myFunction9(myNumber1: number): number {
+    if (Number.isInteger(myNumber1)) {
+        console.log(myNumber1)
+    } else {
+        console.log(Math.ceil(myNumber1))
+    }
+    return (myNumber1)
```
Comment (HeorhSemianenia) [inline hw4/Task_9.ts:12]: no need to return here

File: hw4/Task_3.ts (hunk 12 above)
Comment (DariaBoris) [reply hw4/Task_3.ts]: Ok, will take in mind for the future.
Fixed.

File: hw4/Task_9.ts (hunk 13 above)
Comment (DariaBoris) [reply hw4/Task_9.ts:12]: If I remove it, it throws an error:  'A function whose declared type is neither 'void' nor 'any' must return a value'.

**Pull request merged**